iOS_Learning/ViewController/LogInViewController.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  LogInViewController.swift
+//  iOS_Learning
+//
+//  Created by Vlad_Rosca on 22.03.2023.
+//
+
+import UIKit
+
+class LogInViewController: UIViewController {
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    @IBOutlet weak var loginButton: UIButton!
+    @IBOutlet weak var errorLabel: UILabel!
+    
+    private let loginService = UNLoginService()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Do any additional setup after loading the view.
+        setUpElements()
+    }
+    
+    @IBAction func loginButtonTapped(_ sender: UIButton) {
+        let error = validateFields()
+        
+        if error != nil {
+            showError(error!)
+        } else {
+            login()
+        }
+    }
+    
+    private func transitionToHome() {
```

Comment:
Seems very similar to the one on `SignUpViewController`, why don't we try to take this out of this class's responsibility and leave the upper level decide how to handle the transition? If you create the constructor for the service you can add another parameter (e.g. didFinish: () -> Void, which is a [closure](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/closures/)) and leave the one that displayed this screen to decide what to do on login flow when it's finished.

Reply:
Due to changes regarding dependency injection I don't really see how those part can be implemented wothout further changes.

Reply:
I was suggesting that when the login/signup flow is successfully performed, you should call a closure (that you get on init) rather than doing the same handling/code on both controllers. The one that presented Login or SignUp screens should decide what to do when the flow is finished.